### PR TITLE
[MM-17109] Bump model version to 5.16

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Bumps the model version to 5.16. See https://community.mattermost.com/core/pl/7jadxtfz5fd6pmqb58epoji3ac for further context.

@cpanato I did the PR myself, because there are some plugins relying on this.

### CODE REVIEW

**Comment by cpanato** [comment]
@hanzei please update your PR

**Comment by hanzei** [comment]
I guess the db and app versions are as decoupled as I thought. Tests like https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/upgrade_test.go#L37-L42 are failing now, because the db version doesn't get updated. We could fix this by having a first element of https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/upgrade.go#L22-L66 and using this for the tests. What do you think about this @lieut-data @cpanato ?

**Comment by lieut-data** [comment]
That makes sense to me, @hanzei.

**Comment by hanzei** [comment]
@lieut-data @cpanato Please take a second look

File: store/sqlstore/upgrade.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ import (
 )
 
 const (
+	CURRENT_SCHEMA_VERSION   = VERSION_5_15_0
```
Comment (cpanato) [inline store/sqlstore/upgrade.go]: why is 5.15 and not 5.16?

Comment (hanzei) [reply store/sqlstore/upgrade.go]: This models the current behaviour. The db schema version stays at `5.15` until https://github.com/mattermost/mattermost-server/blob/6e4a8522ff13a015160f4f92133894e7d8679ebe/store/sqlstore/upgrade.go#L748 is commented out, which we will do shortly after the release branch is cut.

Comment (cpanato) [reply store/sqlstore/upgrade.go]: ok

**Comment by cpanato** [comment]
lgtm

**Comment by hanzei** [comment]
@lieut-data Could you please review this? It currently block the demo plugin from being run on `master`.

**Pull request merged**